### DISCUSSION TITLE
Any plans to support JSP?

### DISCUSSION THREAD

**Discussion by slominskir** [github/codeql#17646] [Q&A] [answered]
It appears codeql doesn't currently detect Java Server Pages (JSP) Reflected Cross Site Scripting vulnerabilities.    Any plans to support it?   Rule may roughly involve looking for `${param.*}` in JSP file and if that value isn't one of the three exemption cases then it's vulnerable.  The exemption cases being:

1. isn't located inside a `<c:out value="${?}"/>`  
2. isn't located inside a `${fn:escapeXml(?)}`
3. isn't part of an expression with a function call or operand such as `eq`

**Comment by aibaars** [comment, marked as answer]
Thanks for your question. There has been some work on JSP support in the past, but I don't think it has become a fully supported feature. 

If you're curious you could try running CodeQL with the environment variable `CODEQL_EXTRACTOR_JAVA_JSP=true`. Note that experimental features should typically be avoided for production use.

https://github.com/github/codeql/blob/f7db47b7716d9219ead00676de7e458fe2ce2bda/java/ql/integration-tests/java/java-web-jsp/test.py#L3
